### ISSUE TITLE
Unable to setup Diagnostic Settings for a NSG using Set-AzureRmDiagnosticSetting

### ISSUE THREAD

**Issue by blueelvis** [Azure/azure-powershell#7132]
### Description
I am unable to set diagnostic settings for an NSG using `Set-AzureRmDiagnosticSetting`

Whenever the commandlet runs, all I get is the following 3 warnings and not a single error -
```
WARNING: 21:40:55 - *** The namespace for all the model classes will change from Microsoft.Azure.Management.Monitor.Management.Models 
to Microsoft.Azure.Management.Monitor.Models in future releases.
WARNING: 21:40:55 - *** The namespace for output classes will be uniform for all classes in future releases to make it independent of 
modifications in the model classes.
WARNING: 21:40:55 - The arguments Categories and Timegrains now have aliases Category and Timegrain respectively. The plural names wil
l be removed in future releases.
```

### Script/Steps for Reproduction
```powershell
Set-AzureRmDiagnosticSetting -ResourceId $subnet.NetworkSecurityGroup.Id -StorageAccountId $nsgDiagStorageAccount.Id -Enabled $true -Category $NsgLogCategories -RetentionEnabled $true -RetentionInDays $LogRetentionInDays -WorkspaceId $omsWorkspace.ResourceId -Timegrain "PT1M" -Debug
```

### Module Version
```powershell
Directory: C:\Program Files\WindowsPowerShell\Modules


ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     5.3.0      Azure                               {Get-AzureAutomationCertificate, Get-AzureAutomationConnection, New-AzureAutomationConnection, Remove-AzureAutomationConnection...}
Script     0.5.4      Azure.AnalysisServices              {Add-AzureAnalysisServicesAccount, Restart-AzureAnalysisServicesInstance, Export-AzureAnalysisServicesInstanceLog, Sync-AzureAnalysisServicesInstance}
Script     4.5.0      Azure.Storage                       {Get-AzureStorageTable, New-AzureStorageTableSASToken, New-AzureStorageTableStoredAccessPolicy, New-AzureStorageTable...}
Script     6.7.0      AzureRM
Script     0.6.12     AzureRM.AnalysisServices            {Resume-AzureRmAnalysisServicesServer, Suspend-AzureRmAnalysisServicesServer, Get-AzureRmAnalysisServicesServer, Remove-AzureRmAnalysisServicesServer...}
Script     6.1.3      AzureRM.ApiManagement               {Add-AzureRmApiManagementRegion, Get-AzureRmApiManagementSsoToken, New-AzureRmApiManagementHostnameConfiguration, New-AzureRmApiManagementCustomHostnameConfigu...
Script     0.1.6      AzureRM.ApplicationInsights         {Get-AzureRmApplicationInsights, New-AzureRmApplicationInsights, Remove-AzureRmApplicationInsights, Set-AzureRmApplicationInsightsPricingPlan...}
Script     5.0.4      AzureRM.Automation                  {Get-AzureRMAutomationHybridWorkerGroup, Get-AzureRmAutomationJobOutputRecord, Import-AzureRmAutomationDscNodeConfiguration, Export-AzureRmAutomationDscConfigu...
Script     4.0.8      AzureRM.Backup                      {Backup-AzureRmBackupItem, Enable-AzureRmBackupContainerReregistration, Get-AzureRmBackupContainer, Register-AzureRmBackupContainer...}
Script     4.1.3      AzureRM.Batch                       {Remove-AzureRmBatchAccount, Get-AzureRmBatchAccount, Get-AzureRmBatchAccountKeys, New-AzureRmBatchAccount...}
Script     0.14.5     AzureRM.Billing                     {Get-AzureRmBillingInvoice, Get-AzureRmBillingPeriod, Get-AzureRmEnrollmentAccount}
Script     5.0.3      AzureRM.Cdn                         {Get-AzureRmCdnProfile, Get-AzureRmCdnProfileSsoUrl, New-AzureRmCdnProfile, Remove-AzureRmCdnProfile...}
Script     0.9.8      AzureRM.CognitiveServices           {Get-AzureRmCognitiveServicesAccount, Get-AzureRmCognitiveServicesAccountKey, Get-AzureRmCognitiveServicesAccountSkus, Get-AzureRmCognitiveServicesAccountUsage...
Script     5.5.0      AzureRM.Compute                     {Remove-AzureRmAvailabilitySet, Get-AzureRmAvailabilitySet, New-AzureRmAvailabilitySet, Update-AzureRmAvailabilitySet...}
Script     0.3.7      AzureRM.Consumption                 {Get-AzureRmConsumptionBudget, Get-AzureRmConsumptionMarketplace, Get-AzureRmConsumptionPriceSheet, Get-AzureRmConsumptionReservationDetail...}
Script     0.2.8      AzureRM.ContainerInstance           {New-AzureRmContainerGroup, Get-AzureRmContainerGroup, Remove-AzureRmContainerGroup, Get-AzureRmContainerInstanceLog}
Script     1.0.8      AzureRM.ContainerRegistry           {New-AzureRmContainerRegistry, Get-AzureRmContainerRegistry, Update-AzureRmContainerRegistry, Remove-AzureRmContainerRegistry...}
Script     5.0.3      AzureRM.DataFactories               {Remove-AzureRmDataFactory, Get-AzureRmDataFactoryRun, Get-AzureRmDataFactorySlice, Save-AzureRmDataFactoryLog...}
Script     0.5.8      AzureRM.DataFactoryV2               {Set-AzureRmDataFactoryV2, Update-AzureRmDataFactoryV2, Get-AzureRmDataFactoryV2, Remove-AzureRmDataFactoryV2...}
Script     5.1.2      AzureRM.DataLakeAnalytics           {Get-AzureRmDataLakeAnalyticsDataSource, New-AzureRmDataLakeAnalyticsCatalogCredential, Remove-AzureRmDataLakeAnalyticsCatalogCredential, Remove-AzureRmDataLak...
Script     6.1.2      AzureRM.DataLakeStore               {Get-AzureRmDataLakeStoreTrustedIdProvider, Remove-AzureRmDataLakeStoreTrustedIdProvider, Remove-AzureRmDataLakeStoreFirewallRule, Set-AzureRmDataLakeStoreTrus...
Script     4.0.7      AzureRM.DevTestLabs                 {Get-AzureRmDtlAllowedVMSizesPolicy, Get-AzureRmDtlAutoShutdownPolicy, Get-AzureRmDtlAutoStartPolicy, Get-AzureRmDtlVMsPerLabPolicy...}
Script     5.0.2      AzureRM.Dns                         {Get-AzureRmDnsRecordSet, New-AzureRmDnsRecordConfig, Remove-AzureRmDnsRecordSet, Set-AzureRmDnsRecordSet...}
Script     0.3.7      AzureRM.EventGrid                   {New-AzureRmEventGridTopic, Get-AzureRmEventGridTopic, Set-AzureRmEventGridTopic, New-AzureRmEventGridTopicKey...}
Script     0.6.8      AzureRM.EventHub                    {New-AzureRmEventHubNamespace, Get-AzureRmEventHubNamespace, Set-AzureRmEventHubNamespace, Remove-AzureRmEventHubNamespace...}
Script     4.1.6      AzureRM.HDInsight                   {Get-AzureRmHDInsightJob, New-AzureRmHDInsightSqoopJobDefinition, Wait-AzureRmHDInsightJob, New-AzureRmHDInsightStreamingMapReduceJobDefinition...}
Script     5.1.1      AzureRM.Insights                    {Get-AzureRmMetricDefinition, Get-AzureRmMetric, Remove-AzureRmLogProfile, Get-AzureRmLogProfile...}
Script     3.1.6      AzureRM.IotHub                      {Add-AzureRmIotHubKey, Get-AzureRmIotHubEventHubConsumerGroup, Get-AzureRmIotHubConnectionString, Get-AzureRmIotHubJob...}
Script     5.1.1      AzureRM.KeyVault                    {Add-AzureKeyVaultCertificate, Update-AzureKeyVaultCertificate, Stop-AzureKeyVaultCertificateOperation, Get-AzureKeyVaultCertificateOperation...}
Script     4.1.2      AzureRM.LogicApp                    {Get-AzureRmIntegrationAccountAgreement, Get-AzureRmIntegrationAccountCallbackUrl, Get-AzureRmIntegrationAccountCertificate, Get-AzureRmIntegrationAccount...}
Script     0.18.3     AzureRM.MachineLearning             {Move-AzureRmMlCommitmentAssociation, Get-AzureRmMlCommitmentAssociation, Get-AzureRmMlCommitmentPlanUsageHistory, Remove-AzureRmMlCommitmentPlan...}
Script     0.4.6      AzureRM.MachineLearningCompute      {Get-AzureRmMlOpCluster, Get-AzureRmMlOpClusterKey, Test-AzureRmMlOpClusterSystemServicesUpdateAvailability, Update-AzureRmMlOpClusterSystemService...}
Script     0.2.5      AzureRM.MarketplaceOrdering         {Get-AzureRmMarketplaceTerms, Set-AzureRmMarketplaceTerms}
Script     0.10.2     AzureRM.Media                       {Sync-AzureRmMediaServiceStorageKeys, Set-AzureRmMediaServiceKey, Get-AzureRmMediaServiceKeys, Get-AzureRmMediaServiceNameAvailability...}
Script     6.5.0      AzureRM.Network                     {Add-AzureRmApplicationGatewayAuthenticationCertificate, Get-AzureRmApplicationGatewayAuthenticationCertificate, New-AzureRmApplicationGatewayAuthenticationCer...
Script     5.0.3      AzureRM.NotificationHubs            {Get-AzureRmNotificationHub, Get-AzureRmNotificationHubAuthorizationRules, Get-AzureRmNotificationHubListKeys, Get-AzureRmNotificationHubPNSCredentials...}
Script     5.0.4      AzureRM.OperationalInsights         {New-AzureRmOperationalInsightsAzureActivityLogDataSource, New-AzureRmOperationalInsightsCustomLogDataSource, Disable-AzureRmOperationalInsightsLinuxCustomLogC...
Script     1.0.4      AzureRM.PolicyInsights              {Get-AzureRmPolicyEvent, Get-AzureRmPolicyState, Get-AzureRmPolicyStateSummary}
Script     4.1.8      AzureRM.PowerBIEmbedded             {Remove-AzureRmPowerBIWorkspaceCollection, Get-AzureRmPowerBIWorkspaceCollection, Get-AzureRmPowerBIWorkspaceCollectionAccessKeys, Get-AzureRmPowerBIWorkspace...}
Script     5.4.0      AzureRM.Profile                     {Disable-AzureRmDataCollection, Disable-AzureRmContextAutosave, Enable-AzureRmDataCollection, Enable-AzureRmContextAutosave...}
Script     4.1.6      AzureRM.RecoveryServices            {Get-AzureRmRecoveryServicesBackupProperty, Get-AzureRmRecoveryServicesVault, Get-AzureRmRecoveryServicesVaultSettingsFile, New-AzureRmRecoveryServicesVault...}
Script     4.4.0      AzureRM.RecoveryServices.Backup     {Backup-AzureRmRecoveryServicesBackupItem, Get-AzureRmRecoveryServicesBackupManagementServer, Get-AzureRmRecoveryServicesBackupContainer, Unregister-AzureRmRec...
Script     0.2.8      AzureRM.RecoveryServices.SiteRec... {Edit-AzureRmRecoveryServicesAsrRecoveryPlan, Get-AzureRmRecoveryServicesAsrAlertSetting, Get-AzureRmRecoveryServicesAsrEvent, Get-AzureRmRecoveryServicesAsrFa...
Script     5.0.3      AzureRM.RedisCache                  {Remove-AzureRmRedisCachePatchSchedule, New-AzureRmRedisCacheScheduleEntry, Get-AzureRmRedisCachePatchSchedule, New-AzureRmRedisCachePatchSchedule...}
Script     0.3.8      AzureRM.Relay                       {New-AzureRmRelayNamespace, Get-AzureRmRelayNamespace, Set-AzureRmRelayNamespace, Remove-AzureRmRelayNamespace...}
Script     6.4.0      AzureRM.Resources                   {Get-AzureRmProviderOperation, Remove-AzureRmRoleAssignment, Get-AzureRmRoleAssignment, New-AzureRmRoleAssignment...}
Script     0.16.8     AzureRM.Scheduler                   {Disable-AzureRmSchedulerJobCollection, Enable-AzureRmSchedulerJobCollection, Get-AzureRmSchedulerJobCollection, Get-AzureRmSchedulerJob...}
Script     0.6.9      AzureRM.ServiceBus                  {New-AzureRmServiceBusNamespace, Get-AzureRmServiceBusNamespace, Set-AzureRmServiceBusNamespace, Remove-AzureRmServiceBusNamespace...}
Script     0.3.10     AzureRM.ServiceFabric               {Add-AzureRmServiceFabricApplicationCertificate, Add-AzureRmServiceFabricClientCertificate, Add-AzureRmServiceFabricClusterCertificate, Add-AzureRmServiceFabri...
Script     4.11.1     AzureRM.Sql                         {Get-AzureRmSqlDatabaseTransparentDataEncryption, Get-AzureRmSqlDatabaseTransparentDataEncryptionActivity, Set-AzureRmSqlDatabaseTransparentDataEncryption, Get...
Script     5.0.2      AzureRM.Storage                     {Get-AzureRmStorageAccount, Get-AzureRmStorageAccountKey, New-AzureRmStorageAccount, New-AzureRmStorageAccountKey...}
Script     4.0.8      AzureRM.StreamAnalytics             {Get-AzureRmStreamAnalyticsFunction, Get-AzureRmStreamAnalyticsDefaultFunctionDefinition, New-AzureRmStreamAnalyticsFunction, Remove-AzureRmStreamAnalyticsFunc...
Script     4.0.5      AzureRM.Tags                        {Remove-AzureRmTag, Get-AzureRmTag, New-AzureRmTag}
Script     4.0.9      AzureRM.TrafficManager              {Disable-AzureRmTrafficManagerEndpoint, Enable-AzureRmTrafficManagerEndpoint, Set-AzureRmTrafficManagerEndpoint, Get-AzureRmTrafficManagerEndpoint...}
Script     4.0.5      AzureRM.UsageAggregates             Get-UsageAggregates
Script     5.0.7      AzureRM.Websites                    {Get-AzureRmAppServicePlan, Set-AzureRmAppServicePlan, New-AzureRmAppServicePlan, Remove-AzureRmAppServicePlan...}
Script     1.0.1      Microsoft.PowerShell.Operation.V... {Get-OperationValidation, Invoke-OperationValidation}
Binary     1.0.0.1    PackageManagement                   {Find-Package, Get-Package, Get-PackageProvider, Get-PackageSource...}
Script     3.4.0      Pester                              {Describe, Context, It, Should...}
Script     1.0.0.1    PowerShellGet                       {Install-Module, Find-Module, Save-Module, Update-Module...}
Script     1.2        PSReadline                          {Get-PSReadlineKeyHandler, Set-PSReadlineKeyHandler, Remove-PSReadlineKeyHandler, Get-PSReadlineOption...}
```

### Environment Data

```powershell
PS C:\Users\Pranav> $PSVersionTable

Name                           Value                                                                                                 
----                           -----                                                                                                 
PSVersion                      5.1.17134.228                                                                                         
PSEdition                      Desktop                                                                                               
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                               
BuildVersion                   10.0.17134.228                                                                                        
CLRVersion                     4.0.30319.42000                                                                                       
WSManStackVersion              3.0                                                                                                   
PSRemotingProtocolVersion      2.3                                                                                                   
SerializationVersion           1.1.0.1   
```

### Debug Output
```
DEBUG: 21:40:59 - SetAzureRmDiagnosticSettingCommand begin processing with ParameterSet 'OldSetDiagnosticSetting'.
DEBUG: 21:41:00 - using account id 'aaa@aaa.com'...
WARNING: 21:41:00 - *** The namespace for all the model classes will change from Microsoft.Azure.Management.Monitor.Management.Models 
to Microsoft.Azure.Management.Monitor.Models in future releases.
WARNING: 21:41:01 - *** The namespace for output classes will be uniform for all classes in future releases to make it independent of 
modifications in the model classes.
WARNING: 21:41:02 - The arguments Categories and Timegrains now have aliases Category and Timegrain respectively. The plural names wil
l be removed in future releases.
DEBUG: 21:41:02 - Processing using command line arguments
DEBUG: 21:41:02 - Getting existing diagnostics setting called 'service'
DEBUG: [Common.Authentication]: Authenticating using Account: 'aaa@aaa.com', environment: 'AzureCloud', tenant: '2dfc6
a32-d03c-4504-8d69-f08da326f294'
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 09/05/2018 16:11:02: 7544c1c5-bc89-4b3e-b346-8527db3ea4fc - AcquireTokenHandlerBase: === Token Acquisition started:
	Authority: https://login.microsoftonline.com/2dfc6a32-d03c-4504-8d69-f08da326f294/
	Resource: https://management.core.windows.net/
	ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
	CacheType: Microsoft.Azure.Commands.Common.Authentication.ProtectedFileTokenCache (8 items)
	Authentication Target: User
	

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 09/05/2018 16:11:02:  - TokenCache: Deserialized 8 items to token cache.

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 : 
DEBUG: 09/05/2018 16:11:02: 7544c1c5-bc89-4b3e-b346-8527db3ea4fc - TokenCache: Looking up cache for a token...

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 09/05/2018 16:11:02: 7544c1c5-bc89-4b3e-b346-8527db3ea4fc - TokenCache: An item matching the requested resource was found in th
e cache

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 : 
DEBUG: 09/05/2018 16:11:02: 7544c1c5-bc89-4b3e-b346-8527db3ea4fc - TokenCache: 27.969809185 minutes left until token in cache expires

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 09/05/2018 16:11:02: 7544c1c5-bc89-4b3e-b346-8527db3ea4fc - TokenCache: A matching item (access token or refresh token or both)
 was found in the cache

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 09/05/2018 16:11:02: 7544c1c5-bc89-4b3e-b346-8527db3ea4fc - AcquireTokenHandlerBase: === Token Acquisition finished successfull
y. An access token was retuned:
	Access Token Hash: 7Qeimf36kDK/Z6mCjchwu0XMdo+v4=
	Refresh Token Hash: owWdYZyNEYKC5mB58u7L6nrA6uC54o=
	Expiration Time: 09/05/2018 16:39:00 +00:00
	User Hash: mH8Ub+nMs9cFQb/7j7XmTr72ZzC9Sw=
	

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 09/05/2018 16:11:02: 9100300e-de36-426c-9328-c6d35fc5a1f7 - AcquireTokenHandlerBase: === Token Acquisition started:
	Authority: https://login.microsoftonline.com/2dfc6a32-d03c-4504-8d69-f08da326f294/
	Resource: https://management.core.windows.net/
	ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
	CacheType: Microsoft.Azure.Commands.Common.Authentication.ProtectedFileTokenCache (8 items)
	Authentication Target: User
	

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 09/05/2018 16:11:02:  - TokenCache: Deserialized 8 items to token cache.

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 : 
DEBUG: 09/05/2018 16:11:02: 9100300e-de36-426c-9328-c6d35fc5a1f7 - TokenCache: Looking up cache for a token...

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 09/05/2018 16:11:02: 9100300e-de36-426c-9328-c6d35fc5a1f7 - TokenCache: An item matching the requested resource was found in th
e cache

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 : 
DEBUG: 09/05/2018 16:11:02: 9100300e-de36-426c-9328-c6d35fc5a1f7 - TokenCache: 27.9696594433333 minutes left until token in cache expi
res

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 09/05/2018 16:11:02: 9100300e-de36-426c-9328-c6d35fc5a1f7 - TokenCache: A matching item (access token or refresh token or both)
 was found in the cache

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 : 
DEBUG: 09/05/2018 16:11:02: 9100300e-de36-426c-9328-c6d35fc5a1f7 - AcquireTokenHandlerBase: === Token Acquisition finished successfull
y. An access token was retuned:
	Access Token Hash: FVU7Qeimf36kDK/Z6mCjchwu0XMdo+v4=
	Refresh Token Hash: XowWdYZyNEYKC5mB58u7L6nrA6uC54o=
	Expiration Time: 09/05/2018 16:39:00 +00:00
	User Hash: resn7mH8Ub+nMs9cFQb/7j7XmTr72ZzC9Sw=
	

DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
GET

Absolute Uri:
https://management.azure.com//subscriptions/66d477ee-6241-4568-9e3f-d533bd3a8953/resourceGroups/XZE-E-N-NETWORK-RGP-01/providers/Micro
soft.Network/networkSecurityGroups/NsgTest1/providers/microsoft.insights/diagnosticSettings/service?api-version=2017-05-01-preview

Headers:
x-ms-client-request-id        : 0219f424-7bb9-4ddd-b0c1-99b48637b6d4
accept-language               : en-US

Body:



DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
NotFound

Headers:
Pragma                        : no-cache
Strict-Transport-Security     : max-age=31536000; includeSubDomains
x-ms-request-id               : 00000000-0000-0000-0000-000000000000
x-ms-ratelimit-remaining-subscription-reads: 11976
x-ms-correlation-request-id   : 0b66374c-3ecd-4531-a0e2-bdf72bfb0d40
x-ms-routing-request-id       : CANADAEAST:20180905T161103Z:0b66374c-3ecd-4531-a0e2-bdf72bfb0d40
X-Content-Type-Options        : nosniff
Cache-Control                 : no-cache
Date                          : Wed, 05 Sep 2018 16:11:03 GMT
Server                        : Microsoft-IIS/8.5

Body:
{
  "code": "ResourceNotFound",
  "message": "The diagnostic setting 'service' doesn't exist."
}


Set-AzureRmDiagnosticSetting : Exception type: ErrorResponseException, Message: The diagnostic setting 'service' doesn't exist., 
Code: ResourceNotFound, Status code:NotFound, Reason phrase: Not Found
At C:\Users\Pranav\Source\Repos\NsgAutomation\NetworkWatcher\Set-XzNsgNetworkWatcher.ps1:116 char:9
+         Set-AzureRmDiagnosticSetting -ResourceId $subnet.NetworkSecur ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Set-AzureRmDiagnosticSetting], PSInvalidOperationException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Insights.Diagnostics.SetAzureRmDiagnosticSettingCommand
 
DEBUG: AzureQoSEvent: CommandName - Set-AzureRmDiagnosticSetting; IsSuccess - False; Duration - 00:00:06.8092786; Exception - System.M
anagement.Automation.PSInvalidOperationException: Exception type: ErrorResponseException, Message: The diagnostic setting 'service' do
esn't exist., Code: ResourceNotFound, Status code:NotFound, Reason phrase: Not Found ---> Microsoft.Azure.Management.Monitor.Models.Er
rorResponseException: Operation returned an invalid status code 'NotFound'
   at Microsoft.Azure.Management.Monitor.DiagnosticSettingsOperations.<GetWithHttpMessagesAsync>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Management.Monitor.DiagnosticSettingsOperationsExtensions.<GetAsync>d__1.MoveNext()
   --- End of inner exception stack trace ---
   at Microsoft.Azure.Commands.Insights.MonitorCmdletBase.ExecuteCmdlet()
   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord();
DEBUG: Finish sending metric.
DEBUG: 21:41:07 - SetAzureRmDiagnosticSettingCommand end processing.
DEBUG: 21:41:07 - SetAzureRmDiagnosticSettingCommand end processing.

```

As per the description of the commandlet over [here](https://docs.microsoft.com/en-us/powershell/module/azurerm.insights/set-azurermdiagnosticsetting?view=azurermps-6.8.1), it seems that the `Name` is not required and by default, it gets assumed with `service`

**EDIT - I installed the latest `6.8.1` and please find the debug trace over [here](http://pasted.co/adafb469).**

**Comment by maddieclayton** [comment]
@gucalder Can you take a look at this issue?

**Comment by blueelvis** [comment]
@maddieclayton @gucalder - This issue is fixed in the latest `6.9.1` Azure RM PowerShell release. Related bug -- https://github.com/Azure/azure-powershell/issues/6833

Closing this issue.

**Comment by nipunsahni** [comment]
Hello, I am experiencing the same issue while running the Set-AzureRmDiagnosticSetting command for enabling the diagnostic setting to send data to Log anlytics. There is no error that we receive. I tried finding the Azure RM PowerShell version 6.9.1 but was unable to find one.
